data_manager_select_index_by_path/data_manager/path_name_value_key_manager.py

```diff
@@ -71,22 +78,39 @@ def check_params(self):
         check_tab('index_path', str(self.index_path.absolute().name))
         check_tab('value', self.value)
         check_tab('dbkey', self.dbkey)
+        self.check_extra_columns()
+
+    def check_extra_columns(self):
+        index_properties = self.get_index_properties()
+        index_extra_columns = set(index_properties.get("extra_columns", []))
+        given_extra_columns = self.extra_columns.keys()
+        if index_extra_columns != given_extra_columns:
```

Comment:
Cool, I didn't know dictionary views could be directly compared to set  :+1: . This does make it really incompatible with python 2 (where .`keys()` returns a list), but I suppose that was the intended behavior anyway :wink:

Reply:
Strictly speaking dictionary keys are not ordered so should be a set, right? It is a good thing that the language matches the specification.
Galaxy uses python 2 by default, but galaxy makes it  possible to run your tools in a Conda environment. Unfortunately this was not enabled for data managers, so they use the galaxy python (python 2). A bug was fixed so that data managers also can use conda. So from 18.09 this tool will work in galaxy using python 3.

Reply:
Dictionaries are no longer unordered since the cpython implementation of python 3.6, python 3.7 makes ordered dictionaries mandatory per the language specification. But we're digressing ;-)